dbms/src/Functions/LeastGreatest.h

```diff
@@ -120,5 +122,466 @@ class FunctionVectorizedLeastGreatest : public IFunction
+            }
+
+            if (string_columns.empty()) // fill the result column
+            {
+                block.getByPosition(result).column
+                    = block
+                          .getByPosition(result)
+                          .type->createColumnConst(const_columns[0]->size(), Field(const_res.toString()));
+                return;
+            }
+        }
+
+        // 2. calculate result column for string columns
+        auto string_columns_size = string_columns.size();
+        ColumnString * result_col = nullptr;
+        if (string_columns_size == 1)
+            result_col = const_cast<ColumnString *>(string_columns[0]);
+        else if (string_columns_size >= 2)
+        {
+            std::vector<StringRef> result_string_refs;
+            result_string_refs.resize(string_columns[0]->size());
+            for (size_t i = 1; i < string_columns_size; ++i)
+            {
+                const DB::ColumnString * c0_string;
+                const auto * c1_string = string_columns[i];
+                if (i == 1)
+                {
+                    c0_string = checkAndGetColumn<ColumnString>(string_columns[0]);
+
+                    impl::stringVectorStringVector(
+                        collator,
+                        c0_string->getChars(),
+                        c0_string->getOffsets(),
+                        c1_string->getChars(),
+                        c1_string->getOffsets(),
+                        result_string_refs);
+                }
+                else
+                {
+                    impl::stringRefVectorStringVector(
+                        collator,
+                        result_string_refs,
+                        c1_string->getChars(),
+                        c1_string->getOffsets());
+                }
+            }
+            if (const_columns.empty()) // no const columns, use string columns result
+            {
+                // materialize string columns result
+                auto res_column = ColumnString::create();
+                for (auto & ref : result_string_refs)
+                    res_column->insertData(ref.data, ref.size);
+                block.getByPosition(result).column = std::move(res_column);
+                return;
+            }
+            else
+            {
+                // 3A. merge result columns of const columns and vector columns
+                auto col_str = ColumnString::create();
+                impl::stringRefVectorConstant(
+                    collator,
+                    result_string_refs,
+                    const_res,
+                    col_str->getChars(),
+                    col_str->getOffsets());
+                block.getByPosition(result).column = std::move(col_str);
+                return;
+            }
+        }
+
+        // 3B. merge result columns of const columns and result_col
+        if (string_columns_size == 1 && !const_columns.empty())
```

Comment:
Move to upper `if (string_columns_size == 1)` ? Actually, if `string_columns_size` is 1, `const_columns` must be non-empty because the arguments size is greater than 2.